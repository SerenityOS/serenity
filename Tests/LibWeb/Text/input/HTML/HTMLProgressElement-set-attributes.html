<!DOCTYPE html>
<script src="../include.js"></script>
<script>
    test(() => {
        const progressElement = document.createElement("progress");
        println(`value attribute initial value: ${progressElement.value}`);
        println(`max attribute initial value: ${progressElement.max}`);
        progressElement.value = -1;
        println(`value attribute after setting value attribute to -1: ${progressElement.value}`);
        progressElement.max = -1;
        println(`max attribute after setting max attribute to -1: ${progressElement.max}`);
        progressElement.value = 50;
        println(`value attribute after setting value attribute to 50: ${progressElement.value}`);
        progressElement.max = 100;
        println(`value attribute after setting max attribute to 100: ${progressElement.value}`);
        println(`max attribute after setting max attribute to 100: ${progressElement.max}`);
        progressElement.value = 101;
        println(`value attribute after setting max attribute to 101: ${progressElement.value}`);
        progressElement.value = -1;
        println(`value attribute after setting value attribute to -1: ${progressElement.value}`);
        progressElement.max = 0;
        println(`value attribute after setting max attribute to 0: ${progressElement.value}`);
        println(`max attribute after setting max attribute to 0: ${progressElement.max}`);
    });
</script>
