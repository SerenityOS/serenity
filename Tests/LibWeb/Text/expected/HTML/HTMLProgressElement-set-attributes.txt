value attribute initial value: 0
max attribute initial value: 1
value attribute after setting value attribute to -1: 0
max attribute after setting max attribute to -1: 1
value attribute after setting value attribute to 50: 1
value attribute after setting max attribute to 100: 50
max attribute after setting max attribute to 100: 100
value attribute after setting max attribute to 101: 100
